pennylane/templates/layers/cv_neural_net.py

```diff
@@ -191,28 +154,23 @@ def CVNeuralNetLayers(
         ValueError: if inputs do not have the correct format
     """
 
-    #############
-    # Input checks
-
     wires = Wires(wires)
     repeat = _preprocess(
         theta_1, phi_1, varphi_1, r, phi_r, theta_2, phi_2, varphi_2, a, phi_a, k, wires
     )
 
-    ###############
-
     for l in range(repeat):
-        cv_neural_net_layer(
-            theta_1=theta_1[l],
-            phi_1=phi_1[l],
-            varphi_1=varphi_1[l],
-            r=r[l],
-            phi_r=phi_r[l],
-            theta_2=theta_2[l],
-            phi_2=phi_2[l],
-            varphi_2=varphi_2[l],
-            a=a[l],
-            phi_a=phi_a[l],
-            k=k[l],
-            wires=wires,
-        )
+
+        Interferometer(theta=theta_1[l], phi=phi_1[l], varphi=varphi_1[l], wires=wires)
+
+        r_and_phi_r = [qml.math.expand_dims(r[l], axis=1), qml.math.expand_dims(phi_r[l], axis=1)]
+        r_and_phi_r = qml.math.concatenate(r_and_phi_r, axis=1)
+        broadcast(unitary=Squeezing, pattern="single", wires=wires, parameters=r_and_phi_r)
+
+        Interferometer(theta=theta_2[l], phi=phi_2[l], varphi=varphi_2[l], wires=wires)
+
+        a_and_phi_a = [qml.math.expand_dims(a[l], axis=1), qml.math.expand_dims(phi_a[l], axis=1)]
+        a_and_phi_a = qml.math.concatenate(a_and_phi_a, axis=1)
+        broadcast(unitary=Displacement, pattern="single", wires=wires, parameters=a_and_phi_a)
+
+        broadcast(unitary=Kerr, pattern="single", wires=wires, parameters=k[l])
```

Comment:
This is also much easier to read compared to the previous version with the function call